core/types/block.internal.libevm_test.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	"reflect"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Body_hasOptionalFields(t *testing.T) {
+	t.Parallel()
+
+	var body Body
+
+	assert.False(t, body.hasOptionalFields(), "body has no optional field set")
+
+	v := reflect.ValueOf(&body).Elem()
+	typ := reflect.TypeOf(body)
+	for i := 0; i < v.NumField(); i++ {
+		fieldType := typ.Field(i)
+		rlpTag := fieldType.Tag.Get("rlp")
+		rlpTagFields := strings.Split(rlpTag, ",")
+		if !slices.Contains(rlpTagFields, "optional") {
+			continue
+		}
+
+		field := v.Field(i)
+		before := field.Interface()
+		switch field.Kind() {
+		case reflect.Slice:
```

Comment:
This needs support for `reflect.Pointer` as the optional fields can be struct pointers.

Reply:
We can just add them later on a add-when-needed basis. If it's not defined in the switch, the test would fail